cmd/tier.go

```diff
@@ -116,6 +117,25 @@ func (config *TierConfigMgr) Add(ctx context.Context, tier madmin.TierConfig) er
 	return nil
 }
 
+// Remove removes tier if it is empty.
+func (config *TierConfigMgr) Remove(ctx context.Context, tier string) error {
+	d, err := config.getDriver(tier)
+	if err != nil {
+		return err
+	}
+	if inuse, err := d.InUse(ctx); err != nil {
+		return err
+	} else if inuse {
+		return errTierBackendNotEmpty
```

Comment:
question do we send out valid errors to the client upon failure? instead of HTTP 500 errors?

Reply:
We handle driver errors like `errTierBackendNotEmpty` in `toAdminAPIErr` like here https://github.com/minio/minio/pull/14382/files#diff-415287d4cdaac85f41466937dee8d9e5d2dcdf21a688c35043968fc62b8baa4cR185.

I am working on a separate PR to simplify how we ensure that all tiering errors are API friendly so that we don't implicitly return 500 errors. Today this is ensured through manual testing and is error prone (pun unintended).

Reply:
okay